module/mfrc522_user_command.c

```diff
@@ -63,6 +65,35 @@ static int mem_read(char *answer)
 	return byte_amount;
 }
 
+/**
+ * Write data into the MFRC522's internal memory
+ *
+ * @param data User input to write to the memory
+ * @param data_len Length of the data to write
+ *
+ * @return 0 on success, -1 on error
+ */
+static int mem_write(char *data, u8 data_len)
+{
+	static const char null_bytes[MFRC522_MEM_SIZE] = { 0 };
+
+	if (mfrc522_fifo_write(data, data_len) < 0) {
+		pr_err("[MFRC522] Couldn't write to FIFO\n");
+		return -1;
+	}
+
+	// Complete the remaining slot with NULL bytes
+	mfrc522_fifo_write(null_bytes, MFRC522_MEM_SIZE - data_len);
```

Comment:
We can just do it in one go now that data is filled with 0s and discard ``data_len``

Reply:
Fixed in e6c23bd